### PR TITLE
[5.1] Log Exceptions When A Job Fails

### PR DESCRIPTION
**Pull request by milesj**
Fixes #6203.

### CODE REVIEW

File: src/Illuminate/Queue/Worker.php [hunk 1]
```diff
@@ -195,7 +197,9 @@ public function process($connection, Job $job, $maxTries = 0, $delay = 0)
 	{
 		if ($maxTries > 0 && $job->attempts() > $maxTries)
 		{
-			return $this->logFailedJob($connection, $job);
+			$exception = new RuntimeException(sprintf('Max tries of %d attempted.', $maxTries));
+
```
Comment (taylorotwell) [inline src/Illuminate/Queue/Worker.php]: I'm a little confused by this check to. How would it ever be hit? If a job didn't throw an exception but was released to many times?

Comment (milesj) [inline src/Illuminate/Queue/Worker.php]: I didn't write this check, it was already apart of Laravel, I simply added the exception part.

But yes, that sounds like what it is doing.

Comment (taylorotwell) [inline src/Illuminate/Queue/Worker.php]: OK thanks for info

File: src/Illuminate/Queue/Worker.php [hunk 2]
```diff
@@ -208,8 +212,14 @@ public function process($connection, Job $job, $maxTries = 0, $delay = 0)
 			return ['job' => $job, 'failed' => false];
 		}
 
-		catch (\Exception $e)
+		catch (Exception $e)
 		{
+			// If the job failed and hit the max, let's log it and exit the process early.
+			if ($maxTries > 0 && ($job->attempts() + 1) >= $maxTries)
```
Comment (taylorotwell) [inline src/Illuminate/Queue/Worker.php]: Looking at this check right here. Say my job is on attempt number 2 and max tries is 3. If my job throws an exception and you check attempts() + 1 it's going to log it as a failed job when really it only has 2 attempts?

**Comment by GrahamCampbell** [comment]
This needs rebasing against the latest changes in the master.

**Comment by milesj** [comment]
D:

I can get to this on Monday.

**Comment by GrahamCampbell** [comment]
Great. There's no rush. :)

**Comment by GrahamCampbell** [comment]
Ping @milesj.

**Comment by milesj** [comment]
Oops, totally forgot about this. Thanks for reminding me.

**Comment by taylorotwell** [comment]
Going to hold off on this.

**Comment by milesj** [comment]
Alright :/

**Comment by GrahamCampbell** [comment]
Aw. :( I liked this idea.

**Comment by ashleydw** [comment]
Is there an implementation of this out there somewhere? Jobs currently fail with no log output. A dump of the exception is all that is needed

**Pull request closed without merging**